### PR TITLE
[RFC] Increase support window for v1beta1 CRDs to 1 year

### PR DESCRIPTION
**Pull request by lbernick**
This commit updates the support policy for v1beta1 CRDs to 1 year, to give users more time to migrate off of the v1beta1 API.

v1beta1 is considered a special case because there were no stable APIs available when it was created. To avoid users becoming dependent on future unstable APIs and inhibiting development, the existing policy of 9 months will apply to future beta CRDs.

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- n/a Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- n/a Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Support window for beta CRDs increased to 1 year
```

### CODE REVIEW

**Review by afrittoli — COMMENTED** [review]
Thank you @lbernick 

I thought about this a bit more, and I wonder if perhaps we should leave the current policy for changes to a beta API as it is, and add an extra clause specific about the full removal of an API?

I think 9 months is quite a long time already and I'm afraid that if we make it even longer it will inhibit development when we get to a `v2alpha1` release, because switching to `v2beta1` would already mean 1 year stability commitment, which would result again in us stagnating on the unstable API versions, customers adopting them even if they are unstable, and then the cycle repeating. 

Perhaps instead we could leave the 9 months as they are, and possibly reduce them in future, and add a clause specific for `v1beta1` removal.

**Comment by vdemeester** [comment]
> Thank you @lbernick
> 
> I thought about this a bit more, and I wonder if perhaps we should leave the current policy for changes to a beta API as it is, and add an extra clause specific about the full removal of an API?
> 
> I think 9 months is quite a long time already and I'm afraid that if we make it even longer it will inhibit development when we get to a `v2alpha1` release, because switching to `v2beta1` would already mean 1 year stability commitment, which would result again in us stagnating on the unstable API versions, customers adopting them even if they are unstable, and then the cycle repeating.
> 
> Perhaps instead we could leave the 9 months as they are, and possibly reduce them in future, and add a clause specific for `v1beta1` removal.

That would work for me too (aka both approach LGTM 🙃 )

**Comment by lbernick** [comment]
@afrittoli just to make sure I understand what you're suggesting:
- changes to beta api can be done w/ 9 month warning
- changes to beta features can be done w/ 9 month warning
- a beta crd can't be fully removed until 1 year after it's deprecated?

Or are you suggesting the v1beta1 api is a special case?

FWIW I share your concerns about increasing the support window and having users come to rely on beta apis.

**Comment by lbernick** [comment]
@afrittoli @vdemeester I've updated the proposal to apply to v1beta1 only based on your feedback, PTAL!
@tektoncd/core-maintainers please weigh in by the end of the week; I will merge this PR then if it has approval/lgtm and no pushback.

**Review by vdemeester — APPROVED** [review]

**Review by afrittoli — COMMENTED** [review]
Thanks for the update @lbernick !

/approve

**Pull request merged**
